### PR TITLE
GCP: Container Node Pools

### PR DESCRIPTION
**Pull request by rambleraptor**
##### SUMMARY
Adding support for GCP Container Node Pools

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
gcp_container_node_pool

##### ANSIBLE VERSION
```
2.6
```

##### ADDITIONAL INFORMATION
This was autogenerated using [Magic Modules](https://github.com/GoogleCloudPlatform/magic-modules)

### CODE REVIEW

File: lib/ansible/modules/cloud/google/gcp_container_node_pool.py [hunk 1]
```diff
@@ -0,0 +1,765 @@
...
+                      more information, including usage and the valid values, see:
+                      http://kubernetes.io/v1.1/docs/user-guide/labels.html An
+                      object containing a list of "key": value pairs.'
+                    - 'Example: { "name": "wrench", "mass": "1.3kg", "count": "3" }.'
+                required: false
+            local_ssd_count:
+                description:
+                    - The number of local SSD disks to be attached to the node.
+                    - 'The limit for this value is dependant upon the maximum number of disks available
+                      on a machine per zone. See:  https://cloud.google.com/compute/docs/disks/   local-ssd#local_ssd_limits  for
```
Comment (s-hertel) [inline lib/ansible/modules/cloud/google/gcp_container_node_pool.py]: Extra spaces before and after `local-ssd#local_ssd_limits`

File: lib/ansible/modules/cloud/google/gcp_container_node_pool.py [hunk 2]
```diff
@@ -0,0 +1,765 @@
...
+                description:
+                    - Size of the disk attached to each node, specified in GB. The smallest allowed disk
+                      size is 10GB. If unspecified, the default disk size is 100GB.
+                required: false
+            oauth_scopes:
+                description:
+                    - The set of Google API scopes to be made available on all of the node VMs under the
+                      "default" service account.
+                    - 'The following scopes are recommended, but not required, and by default are not
+                      included:  https://www.googleapis.com/auth/compute is required for mounting persistent
```
Comment (s-hertel) [inline lib/ansible/modules/cloud/google/gcp_container_node_pool.py]: URLs should be in U() to be formatted properly, so `U(https://www.googleapis.com/auth/compute)`. Same for any of the following links.

File: lib/ansible/modules/cloud/google/gcp_container_node_pool.py [hunk 3]
```diff
@@ -0,0 +1,765 @@
...
+                      http://kubernetes.io/v1.1/docs/user-guide/labels.html An
+                      object containing a list of "key": value pairs.'
+                    - 'Example: { "name": "wrench", "mass": "1.3kg", "count": "3" }.'
+                returned: success
+                type: dict
+            local_ssd_count:
+                description:
+                    - The number of local SSD disks to be attached to the node.
+                    - 'The limit for this value is dependant upon the maximum number of disks available
+                      on a machine per zone. See:  https://cloud.google.com/compute/docs/disks/   local-ssd#local_ssd_limits  for
```
Comment (s-hertel) [inline lib/ansible/modules/cloud/google/gcp_container_node_pool.py]: Extra spaces before and after `local-ssd#local_ssd_limits` here too.

**Review by s-hertel — APPROVED** [review]
Looks good, just needs some URL formatting.

File: lib/ansible/modules/cloud/google/gcp_container_node_pool.py [hunk 4]
```diff
@@ -0,0 +1,765 @@
...
+author: Google Inc. (@googlecloudplatform)
+requirements:
+    - python >= 2.6
+    - requests >= 2.18.4
+    - google-auth >= 1.3.0
+options:
+    state:
+        description:
+            - Whether the given object should exist in GCP
+        required: true
```
Comment (s-hertel) [inline lib/ansible/modules/cloud/google/gcp_container_node_pool.py]: Shouldn't have `required: true` for things that have a default.

**Comment by rambleraptor** [comment]
PTAL

**Review by ryansb — APPROVED** [review]

**Pull request merged**